pkg/roachprod/promhelperclient/client.go

```diff
@@ -269,30 +245,59 @@ func buildCreateRequest(nodes map[int][]*NodeInfo, insecure bool) (io.Reader, er
 	if err != nil {
 		return nil, err
 	}
-	b, err := json.Marshal(&instanceConfigRequest{Config: string(cb), Insecure: insecure})
+	b, err := json.Marshal(&instanceConfigRequest{
+		Config: string(cb),
+	})
 	if err != nil {
 		return nil, err
 	}
 	return bytes.NewReader(b), nil
 }
 
-// getToken gets the Authorization token for grafana
-func (c *PromClient) getToken(
-	ctx context.Context, forceFetchCreds bool, l *logger.Logger,
-) (string, error) {
-	if strings.HasPrefix(c.promUrl, "http:/") {
-		// no token needed for insecure URL
-		return "", nil
-	}
-	// Read in the service account key and audience, so we can retrieve the identity token.
-	if credSrc, err := roachprodutil.SetServiceAccountCredsEnv(ctx, forceFetchCreds); err != nil {
-		return "", err
-	} else {
-		l.Printf("Prometheus credentials obtained from %s", credSrc)
-	}
-	token, err := roachprodutil.GetServiceAccountToken(ctx, c.newTokenSource)
-	if err != nil {
-		return "", err
-	}
-	return fmt.Sprintf("Bearer %s", token), nil
+// instanceConfigRequest is the HTTP request received for generating instance config
+type instanceConfigRequest struct {
+	//Config is the content of the yaml file
+	Config   string `json:"config"`
+	Insecure bool   `json:"insecure"`
+}
+
+// CCParams are the params for the cluster configs
+type CCParams struct {
+	Targets []string          `yaml:"targets"`
+	Labels  map[string]string `yaml:"labels"`
+}
+
+// NodeInfo contains the target and labels for the node
+type NodeInfo struct {
+	Target       string            // Name of the node
+	CustomLabels map[string]string // Custom labels to be added to the cluster config
+}
+
+type Option interface {
```

Comment:
Defining structs that implement the `Option` interface for every option seems unnecessary to me.

Something like: 

```
type Option interface {
	apply(*PromClient)
}

type OptionFunc func(c *PromClient)

func (o OptionFunc) apply(c *PromClient) { o(c) }

func WithIAPTokenSource(tokenSource roachprodutil.IAPTokenSourceIface) OptionFunc {
	return func(c *PromClient) {
		c.client = tokenSource.GetHTTPClient()
	}
}

func WithCustomURL(url string) OptionFunc {
	return func(c *PromClient) {
		c.promUrl = url
		c.disabled = false
	}
}
```

Should also work?

Reply:
Right, much better! Thanks!